File: source/net/auoeke/cin/lexer/lexeme/DelimiterLexeme.java
package net.auoeke.cin.lexer.lexeme;

import net.auoeke.cin.lexer.error.SyntaxError;

public final class DelimiterLexeme extends Lexeme {
    public final Token token;

    public DelimiterLexeme(int line, int column, Token token, SyntaxError error) {
        super(line, column, error);

        this.token = token;
    }

    public DelimiterLexeme(int line, int column, Token token) {
        this(line, column, token, null);
    }

    @Override public Token token() {
        return this.token;
    }

    @Override public String toString() {
        return String.valueOf(this.token().character());
    }
}

File: src/net/auoeke/cin/lexer/lexeme/DelimiterLexeme.kt
package net.auoeke.cin.lexer.lexeme

import net.auoeke.extensions.string

class DelimiterLexeme(line: Int, column: Int, val value: Char) : Lexeme(line, column) {
    override val type: Type get() = Type.DELIMITER

    override fun toString(): String = value.string
}
